### ISSUE TITLE
Fallback with Non-Yubikey passphrase not working

### ISSUE THREAD

**Issue by nilmerg** [cornelinux/yubikey-luks#47]
I've put the following as suggested in the README into my /etc/crypttab:

```
...,keyscript=/usr/share/yubikey-luks/ykluks-keyscript
```

The Yubikey integration works fine, fortunately. Though, it won't accept my other passphrase which has nothing to do with the Yubikey.

Plugging out the Yubikey does not help. (I did a cold boot after that.)
Also, just pressing Enter without having anything typed already challenges the Yubikey. It's still showing `Please insert yubikey and press enter or enter a valid passphrase` in this case.

Any help?

I took a look in the keyscript already, but without knowing anything about the integration with LUKS I can't tell if this is a bug or a mis-configuration.

Environment:
Ubuntu 18.04
yubikey-luks: master

**Comment by Vincent43** [comment]
What happens whet you enter a passphrase without YubiKey?

**Comment by nilmerg** [comment]
I already tried that.

> Plugging out the Yubikey does not help. (I did a cold boot after that.)

Though, I previously had problems (Mostly due to non-ascii chars) with LUKS and my passwords so I simply set `test` as a password. While the Yubikey was plugged in it did not work, but with it plugged out it did.
The previous password was a hex key of 32 chars length. I may have mis-typed some chars of course, but surely not every single time. (And I tried it way more than once.) Also a hex key does only consist of ascii chars so...

Anyway, my very first password (Before I performed the Yubikey integration) had 8 chars but also had some special chars. It worked fine. May it be the length? I just did another test with a 16 char long ascii password and didn't work as well. :thinking:

**Comment by Vincent43** [comment]
I asked what was the result :smile:. Was there any message?

**Comment by nilmerg** [comment]
Oh, oops. Well, the usual message about *a bad password or options*.

Did another test yesterday with a (very difficult to mis-type) password of 30 chars length. Which worked pretty well.

I guess it's a simple case of being unable to type the correct password. :roll_eyes: 
Sorry about all the hassle and thanks for stepping by. If I come to another conclusion I'll re-open the issue.

**Comment by Vincent43** [comment]
It may be a problem with keyboard layout being different on your system and in initramfs so you may get different chars printed under the same key in those environments.

**Comment by nilmerg** [comment]
A password consisting only of hexadecimal characters shouldn't be a problem in this case. Which it was.

I did some further tests yesterday and discovered that it's been indeed my fault. Though, not because I typed my passwords all wrong. Some passwords, especially the *long* ones ;), I've temporarily written to a file (without the nasty newline, of course) and then used `--key-file` of `cryptsetup luksChangeKey` instead of typing them interactively.

Well, I thought it wouldn't make any difference. That it's just another variant of defining the desired password. How wrong I have been... Once I've typed my *long* password interactively I can decrypt my disk on boot pretty fine.
